.github/actions/templates/setEnvironmentVariables/action.yml

```diff
@@ -0,0 +1,71 @@
+#########################################################
+##
+## This composite action contains the logic to read a YAML file containing a list (dictionary) of key-value pairs and convert each key as an environment variable on the runner.
+## Example YAML File content:
+## listName:
+##  key1: 'value1'
+##  key2: 'value2'
+#########################################################
+##
+##-------------------------------------------##
+## ACTION PARAMETERS                         ##
+##-------------------------------------------##
+##
+##   |=======================================================================================================================================================================================================|
+##   | Parameter                | Required | Default | Description                                                                                      | Example                                            |
+##   |--------------------------|----------|---------|--------------------------------------------------------------------------------------------------|----------------------------------------------------|
+##   | variablesPath            | true     | ''      | The path to the YAML file that contains the key value pairs for environment variables            | './global.variables.yml'                           |
+##   |=======================================================================================================================================================================================================|
+##
+##---------------------------------------------##
+
+name: 'Set Environment Variables'
+description: 'Set Environment Variables'
+
+inputs:
+  variablesPath:
+    description: 'The path to the YAML file that contains the key value pairs for environment variables'
+    required: true
+
+runs:
+  using: 'composite'
+  steps:
+    # [Setup] task(s)
+    # ---------------
+    - name: 'Setup agent'
+      shell: pwsh
+      run: |
+        # Grouping task logs
+        Write-Output "::group::Setup agent"
+
+        # Load used functions
+        . (Join-Path $env:GITHUB_WORKSPACE 'utilities' 'pipelines' 'sharedScripts' 'Set-EnvironmentOnAgent.ps1')
+
+        # Define PS modules to install on the runner
+        $Modules = @(
+            @{ Name = 'powershell-yaml'; Version = '0.4.2'}
+        )
+
+        # Set agent up
+        Set-EnvironmentOnAgent -PSModules $Modules
+
+        Write-Output "::endgroup::"
+
+    # [Set environment variables] task(s)
+    # ---------------
+    - name: 'Set environment variables'
+      id: set-env-variables
+      shell: pwsh
+      run: |
+        # Grouping task logs
+        Write-Output "::group::Set environment variables"
+
+        # Load used functions
+        . (Join-Path $env:GITHUB_WORKSPACE 'utilities' 'pipelines' 'sharedScripts' 'Add-YamlListToFile.ps1')
+
+        # Convert YAML Variable File to Environment Variables
```

Comment:
I'd appreciate if we could add the 'Invoked task with' snippet too

Reply:
done